src/python_testing/TC_DISHM_2_1.py

```diff
@@ -0,0 +1,159 @@
+#    limitations under the License.
+#
+
+import logging
+
+import chip.clusters as Clusters
+from matter_testing_support import MatterBaseTest, async_test_body, default_matter_test_main, type_matches
+from mobly import asserts
+
+
+class TC_DISHM_2_1(MatterBaseTest):
+
+    async def read_mode_attribute_expect_success(self, endpoint, attribute):
+        cluster = Clusters.Objects.DishwasherMode
+        return await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attribute)
+
+    async def send_change_to_mode_cmd(self, newMode) -> Clusters.Objects.DishwasherMode.Commands.ChangeToModeResponse:
+        ret = await self.send_single_cmd(cmd=Clusters.Objects.DishwasherMode.Commands.ChangeToMode(newMode=newMode), endpoint=self.endpoint)
+        asserts.assert_true(type_matches(ret, Clusters.Objects.DishwasherMode.Commands.ChangeToModeResponse),
+                            "Unexpected return type for ChangeToMode")
+        return ret
+
+    @async_test_body
+    async def test_TC_DISHM_2_1(self):
+
+        asserts.assert_true('PIXIT.DISHM.MODE_CHANGE_OK' in self.matter_test_config.global_test_params,
+                            "PIXIT.DISHM.MODE_CHANGE_OK must be included on the command line in "
+                            "the --int-arg flag as PIXIT.DISHM.MODE_CHANGE_OK:<mode id>")
+        asserts.assert_true('PIXIT.DISHM.MODE_CHANGE_FAIL' in self.matter_test_config.global_test_params,
+                            "PIXIT.DISHM.MODE_CHANGE_FAIL must be included on the command line in "
+                            "the --int-arg flag as PIXIT.DISHM.MODE_CHANGE_FAIL:<mode id>")
+
+        self.endpoint = self.user_params.get("endpoint", 1)
+        logging.info("This test expects to find this cluster on endpoint 1")
+
+        self.modeOk = self.matter_test_config.global_test_params['PIXIT.DISHM.MODE_CHANGE_OK']
+        self.modeFail = self.matter_test_config.global_test_params['PIXIT.DISHM.MODE_CHANGE_FAIL']
+
+        asserts.assert_true(self.check_pics("DISHM.S.A0000"), "DISHM.S.A0000 must be supported")
+        asserts.assert_true(self.check_pics("DISHM.S.A0001"), "DISHM.S.A0001 must be supported")
+        asserts.assert_true(self.check_pics("DISHM.S.C00.Rsp"), "DISHM.S.C00.Rsp must be supported")
+        asserts.assert_true(self.check_pics("DISHM.S.C01.Tx"), "DISHM.S.C01.Tx must be supported")
+
+        attributes = Clusters.DishwasherMode.Attributes
+
+        self.print_step(1, "Commissioning, already done")
+
+        self.print_step(2, "Read SupportedModes attribute")
+        supported_modes = await self.read_mode_attribute_expect_success(endpoint=self.endpoint, attribute=attributes.SupportedModes)
+
+        logging.info("SupportedModes: %s" % (supported_modes))
+
+        asserts.assert_greater_equal(len(supported_modes), 2, "SupportedModes must have at least two entries!")
+
+        supported_modes_dut = [m.mode for m in supported_modes]
```

Comment:
Test plan says to verify that `self.modeFail` is in `supported_modes_dut`, no?  Though it says that before it sets the value of `supported_modes_dut`...